src/main/java/Parser.java

```
@@ -0,0 +1,162 @@
+            } catch  (NumberFormatException e) {
+                throw new DukeException("☹ OOPS!!! Invalid index provided!");
+            }
+        } else {
+            throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+
+    }
+
+    private static Command getFunctionalCommand(String command) throws DukeException {
+        if (command.equals(EXIT_COMMAND)) {
+            return new ExitCommand();
+        } else if (command.equals(LIST_COMMAND)) {
+            return new ListCommand();
+        } else  if (command.startsWith(FIND_COMMAND)) {
+            return new FindCommand(command.substring(5));
+        } else {
+            throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+
+    }
+
+    /**
+     * To format time of user input
+     *
+     * @param input user input
+     * @return formatted time string
+     * @throws DukeException if invalid time format
+     */
+    private static String formatTime(String input) throws DukeException {
+        List<String> formatStrings = Arrays.asList("d/M/y HHmm", "d/M/y", "d/M");
+        List<String> outputFormats = Arrays.asList("MMMMM dd, yyyy, h:mm a", "MMMMM dd, yyyy", "MMMMM dd");
+
+        String[] temp = input.split("/",2);
+        assert temp.length == 2;
+        StringBuilder fullCommand = new StringBuilder(temp[0]);
+        StringBuilder rawDate = new StringBuilder(temp[1]);
+
+        Date date = null;
+        for (int i = 0; i < formatStrings.size(); i++) {
+            try {
+                DateFormat df = new SimpleDateFormat(formatStrings.get(i));
+                df.setLenient(false);
+                date = df.parse(rawDate.toString().substring(3));
+                DateFormat formatter = new SimpleDateFormat(outputFormats.get(i));
+                fullCommand.append("   ").append(formatter.format(date));
+                break;
+
+            } catch (ParseException e) {
```

Try not to leave empty catch blocks in your code! 😄